### PR TITLE
IRIS-927-align-java-fps-options

### PR DESCRIPTION
**Pull request by paweltomaszewskisaucelabs**
## Description
Add new fps options into java bindings

## Types of Changes
Configuration change

### CODE REVIEW

File: visual-java/src/main/java/com/saucelabs/visual/VisualApi.java [hunk 1]
```diff
@@ -190,6 +201,15 @@ public void setCaptureDom(Boolean captureDom) {
     this.captureDom = captureDom;
   }
 
+  /**
+   * Enables full page screenshots
+   *
+   * @param fullPageScreenshotConfig activation of DOM Capture.
```
Comment (FriggaHel) [inline visual-java/src/main/java/com/saucelabs/visual/VisualApi.java:207]: Not sure to understand what that means.

File: visual-java/src/main/java/com/saucelabs/visual/VisualApi.java [hunk 2]
```diff
@@ -370,7 +390,10 @@ public void sauceVisualCheck(String snapshotName, CheckOptions options) {
       input.setClipSelector(clipSelector);
     }
 
-    input.setFullPageConfig(options.getFullPageScreenshotConfig());
+    FullPageScreenshotConfig fullPageScreenshotConfig = Optional.ofNullable(options.getFullPageScreenshotConfig()).orElse(this.fullPageScreenshotConfig);
+    if (fullPageScreenshotConfig != null) {
+      input.setFullPageConfig(fullPageScreenshotConfig);
+    }
```
Comment (FriggaHel) [inline visual-java/src/main/java/com/saucelabs/visual/VisualApi.java:396]: Wouldn't something like this be simplier ?
```suggestion
    input.setFullPageConfig(
      Optional.ofNullable(options.getFullPageScreenshotConfig()).orElse(this.fullPageScreenshotConfig)
    );
```

Comment (kb-kerem) [reply visual-java/src/main/java/com/saucelabs/visual/VisualApi.java:396]: I think the intention is not to set it on `input` if `this.fullPageScreenshotConfig` is null. It doesn't read great, but it follows the way it was done `captureDom`

File: visual-java/src/main/java/com/saucelabs/visual/VisualApi.java [hunk 3]
```diff
@@ -190,6 +201,15 @@ public void setCaptureDom(Boolean captureDom) {
     this.captureDom = captureDom;
   }
 
+  /**
+   * Enables full page screenshots
+   *
+   * @param fullPageScreenshotConfig activation of DOM Capture.
+   */
+  public void setFullPageScreenshotConfig(FullPageScreenshotConfig fullPageScreenshotConfig) {
```
Comment (kb-kerem) [inline visual-java/src/main/java/com/saucelabs/visual/VisualApi.java:209]: I see that this is aligned with the way captureDom is handled but in CheckOptions there are two functions provided for enabling FPS called `enableFullPageScreenshots()` and `enableFullPageScreenshots(FullPageScreenshotConfig config)`. This arguably reads nicer (as it's a user facing API) and provides a way to enable basic FPS without passing a config. I suggest we do the same here

File: visual-java/src/main/java/com/saucelabs/visual/VisualApi.java (hunk 1 above)
Comment (paweltomaszewskisaucelabs) [reply visual-java/src/main/java/com/saucelabs/visual/VisualApi.java]: good catch!

File: visual-java/src/main/java/com/saucelabs/visual/VisualApi.java (hunk 3 above)
Comment (paweltomaszewskisaucelabs) [reply visual-java/src/main/java/com/saucelabs/visual/VisualApi.java]: Changed. Good idea, thanks.

File: visual-java/src/main/java/com/saucelabs/visual/model/FullPageScreenshotConfig.java [hunk 4]
```diff
@@ -6,29 +6,68 @@
...
 
+    public Builder withHideScrollBars(Boolean hideScrollBars) {
+      this.hideScrollBars = hideScrollBars;
+      return this;
+    }
+
+    public Builder withScrollLimit(int scrollLimit) {
+      if (scrollLimit > 0 && scrollLimit <= 10) {
+        this.scrollLimit = scrollLimit;
+      } else {
```
Comment (FriggaHel) [inline visual-java/src/main/java/com/saucelabs/visual/model/FullPageScreenshotConfig.java:57]: This should be enforced only by the backend.

If for some reason we raise the scrollLimit, we will have to update the client which is unnecessary burden.

Client should only check that this is a positive value.

**Review by FriggaHel — APPROVED** [review]

**Pull request merged**